core/src/main/scala/org/apache/spark/api/r/RRunner.scala

```diff
@@ -152,7 +152,7 @@ private[spark] class RRunner[U](
           dataOut.writeInt(mode)
 
           if (isDataFrame) {
-            SerDe.writeObject(dataOut, colNames)
+            SerDe.writeObject(dataOut, colNames, jvmObjectTracker = null)
```

Comment:
I think this change is fine as we are only writing out an array of string -- the other thing we can do is replace the code here and not call SerDe -- but instead just write out the code for writing an array of strings.